### DISCUSSION TITLE
Adjusting texture compression strength [gltfpack]

### DISCUSSION THREAD

**Discussion by Nolram12345** [zeux/meshoptimizer#668] [Q&A] [answered]
Hello! I currently have a very large GLB file I'm trying to compress down, with a focus on compressing for in-memory usage (disk size is less relevant).
I've tried using the -cc and -tc options. These did indeed bring down the VRAM usage by half and the file size from 1.2GB to 260mb, however textures now appear a fair bit too compressed for my liking in-renderer. 

Is there a way to adjust compression strength or compressor used?

P.S. - Is it normal for a 260mb GLB with only a handful compressed 4k textures to expand into 7GB in VRAM? How can I determine what is merely asset size, and what is renderer inefficiency?

**Comment by Nolram12345** [comment]
Here's a visual comparison between
TC OFF:
![gltf_viewer_3C6HZswDNI](https://github.com/zeux/meshoptimizer/assets/37701550/58ab46bf-2807-4eca-a231-308c492a8820)

TC ON:
![gltf_viewer_dP5jO1RSw5](https://github.com/zeux/meshoptimizer/assets/37701550/2d22a9e9-1ce4-47bf-a941-5b0ca30c4418)

The exact position shown is a bit different, but the visual issues should be apparent.

**Reply by zeux** [reply, marked as answer]
You can try:

1. `-tq 10` will try using the maximum quality for the default compression algorithm (ETC1S). This is unlikely to be enough but is the first thing to try.
2. `-tu` will use UASTC which is a much higher quality texture compression; it will need ~double the VRAM upon loading
3. You can try to use `-tu normal` which will use UASTC for normal maps but the (default) ETC1S for other textures. This can act as a balance between size & quality impact from using UASTC everywhere.

You can also specify -tq with UASTC but the defaults are probably good enough. If UASTC helps but the size is too significant, reducing the quality (eg `-tq 3 -tu`) might also be worth a try.

**Reply by Nolram12345** [reply]
Thank you! I'll experiment with that.
